src/classes/BDI_DataImportService.cls

```diff
@@ -41,16 +41,20 @@ global with sharing class BDI_DataImportService {
     public static final String FM_HELP_TEXT = 'Help Text';
     public static final String FM_DATA_IMPORT_FIELD_MAPPING = 'Data Import Field Mapping';
 
-    private BDI_FieldMapping fieldMapper;
+    public BDI_FieldMapping fieldMapper;
 
     /*******************************************************************************************************
     * @description constructor
     * @param isDryRun whether to run in Dry Run mode or commit records.
     */
     public BDI_DataImportService(Boolean isDryRun, BDI_FieldMapping fieldMapper) {
-        contactService = new BDI_ContactService(this);
         this.isDryRun = isDryRun;
         this.fieldMapper = fieldMapper;
+
+        contactService = new BDI_ContactService(this);
+        if (fieldMapper != null && String.valueOf(fieldMapper).split(':')[0] == 'BDI_FieldMappingCustomMetadata'){
```

Comment:
I see why we need to check if we can support the additional objects here, but could we at least use "instanceOf" rather than a string comparison?

Even better is if we can find a slicker way to handle this behavior consistently. Help text mapping is inconsistent with any of the additional objects because it doesn't have the predecessor relationship and other additional metadata available, right? This wants something less brute force I think.

Reply:
I think I can change the method of class comparison, I forget the reason I went this way.  In general though there is no way to additional object mapping with helptext so there needs to be a switch somewhere at some level.

Reply:
What about a method in the base class, "Boolean supportsAdditionalObjects()"? Then the help text variant returns false and the CMT variant returns true?

Reply:
We could do that, but we will still need to check the class type unless we move the map variables that contain the mappings into the base field mapping class (see my other comment for more detail)